### DISCUSSION TITLE
How easy is it to use Next.js with non-Node API server?

### DISCUSSION THREAD

**Discussion by Dehorser** [vercel/next.js#11589] [Help]
I'm interested in using Next.js for SSR, but I'm not a big fan of JavaScript for API code in general. 

1. How easy is it to use Next.js with an API server in Ruby or Flask?

2. What functionality is lost by using my own API server?

**Comment by timneutkens** [comment]
You can fetch any API endpoint to get data, so it can be written in any language you like. Using `getStaticProps` or `getServerSideProps`: https://nextjs.org/docs/basic-features/data-fetching

**Reply by Dehorser** [reply]
Awesome. Is there any risk of clashing with `pages/api`?

**Reply by timneutkens** [reply]
No, there is no risk in not using it 👍
